classes/rule.php

```diff
@@ -136,6 +136,30 @@ public function load_from_form($form) {
 
         $this->data_save();
     }
+    /**
+     * Load a rule from a trigger object.
+     * @param $trigger
+     */
+    public function load_rule_from_trigger($trigger) {
```

Comment:
First thing I'd look at doing here is make this a static method. That way we'll be able to call it without first having to instantiate a rule object that we won't use.